src/engine/enginebufferscale.h

```diff
@@ -65,17 +48,27 @@ class EngineBufferScale : public QObject {
     }
 
     // Set the desired output sample rate.
-    virtual void setSampleRate(int iSampleRate) {
-        m_iSampleRate = iSampleRate;
+    virtual void setSampleRate(SINT iSampleRate);
+
+    const Mixxx::AudioSignal& getAudioSignal() const {
+        return m_audioSignal;
     }
 
     // Called from EngineBuffer when seeking, to ensure the buffers are flushed */
     virtual void clear() = 0;
     // Scale buffer
-    virtual double getScaled(CSAMPLE* pOutput, const int iBufferSize) = 0;
+    // Returns the virtual number of sample frames that have been read.
```

Comment:
The term virtual is misleading here since the frames of the file are real.  

"Returns the number of frames that have bean read from the unscaled input buffer. ..."